New Entry: Cloud FaaS (#945)
Codecademy/docs: content/cloud-computing/concepts/faas/faas.md

---
Title: 'FaaS'
Description: 'Function as a service (FaaS) is a cloud computing model that offers the ability to develop applications without building and maintaining the typical infrastructure needed for application development and deployment.'
Subjects:
  - 'Computer Science'
  - 'Networking'
Tags:

>   - 'Cloud'
>   - 'Functions'
>   - 'Development'

Reviewer: ```suggestion
  - 'Cloud'
  - 'Functions'
  - 'Development'
  - 'Infrastructure'
```

CatalogContent:
  - 'foundations-of-cloud-computing'
  - 'paths/back-end-engineer-career-path'
---

**Function as a service (FaaS)** is a cloud computing model that offers the ability to develop applications without building and maintaining the typical infrastructure needed for application development and deployment. The application functionality is developed and hosted on the cloud resulting in a "serverless" architecture.

## History

FaaS was first offered by start-ups like PiCloud around 2010. It has since been offered by large vendors such as Google (Google Cloud Functions) and Microsoft (Microsoft Azure Functions).

## Advantages and Disadvantages

> The main difference between FaaS and PaaS services is that FaaS services hide the server details from the user. The user is not paying for server processes being run on the cloud, but instead paying for function execution time. Because FaaS does not require a server process to be constantly run, the user does not pay for any process idle time. The disadvantage to this is that initial requests might take several seconds to respond (though caching may reduce subsequent requests to milliseconds). The advantage is a lower cost and higher scalability.

Reviewer: ```suggestion
The main difference between the FaaS and platform as a service (PaaS) models is that FaaS services hide the [server](https://www.codecademy.com/resources/docs/general/server) details from the user. Instead of paying for server processes being run on the cloud, the user is paying for function execution time. Because FaaS does not require a server process to be constantly run, the user does not pay for any process idle time. The disadvantage to this is that initial requests might take several seconds to respond (though [caching](https://www.codecademy.com/resources/docs/general/cache) may reduce subsequent requests to milliseconds). The advantage is a lower cost and higher scalability.
```

---
Title: 'FaaS'
Description: 'Function as a service (FaaS) is a cloud computing model that offers the ability to develop applications without building and maintaining the typical infrastructure needed for application development and deployment.'
Subjects:
  - 'Computer Science'

>   - 'Networking'

Reviewer 2: Should this count as networking just because its on the internet?

Tags:
  - 'Cloud'
  - 'Functions'
  - 'Development'
  - 'Infrastructure'
CatalogContent:
  - 'foundations-of-cloud-computing'
  - 'paths/back-end-engineer-career-path'
---

> **Function as a service (FaaS)** is a cloud computing model that offers the ability to develop applications without building and maintaining the typical infrastructure needed for application development and deployment. The application functionality is developed and hosted on the cloud resulting in a "serverless" architecture.

Reviewer 2: I don't think the term "serverless" is made clear here. Maybe mention like "The lack of needing to configure and maintain servers for these architecture has led to them being called "serverless"

Reviewer: How does this read/sound?

```suggestion
**Function as a service (FaaS)** is a cloud computing model that offers the ability to develop applications without building and maintaining the typical infrastructure needed for application development and deployment. The application functionality is developed and hosted on the cloud. This removes the need for the configuration and maintenance of a server, resulting in a "serverless" architecture.
```

Reviewer: This would bring the first use of "server" at the beginning of the entry, hence the new Docs link.

## History

> FaaS was first offered by start-ups like PiCloud around 2010. It has since been offered by large vendors such as Google (Google Cloud Functions) and Microsoft (Microsoft Azure Functions).

Reviewer 2: Think AWS Lambda should be included too

Reviewer: ```suggestion
FaaS was first offered by start-ups like PiCloud around 2010. It has since been offered by large vendors such as Amazon (AWS Lambda), Google (Google Cloud Functions), and Microsoft (Microsoft Azure Functions).
```

## Advantages and Disadvantages

> The main difference between the FaaS and platform as a service (PaaS) models is that FaaS services hide the [server](https://www.codecademy.com/resources/docs/general/server) details from the user. Instead of paying for server processes being run on the cloud, the user is paying for function execution time. Because FaaS does not require a server process to be constantly run, the user does not pay for any process idle time. The disadvantage to this is that initial requests might take several seconds to respond (though [caching](https://www.codecademy.com/resources/docs/general/cache) may reduce subsequent requests to milliseconds). The advantage is a lower cost and higher scalability.

Reviewer 2: Would consider breaking this into sections rather than one big paragraph. Also the content goes "Advantage->Disadvantage->Advantage" the flow of which I don't think makes sense.

Reviewer: Based on Nik's [prior comment](https://github.com/Codecademy/docs/pull/945#discussion_r964002503), I'd also suggest removing the link to "server" here.

---
Title: 'FaaS'

> Description: 'Function as a service (FaaS) is a cloud computing model that offers the ability to develop applications without building and maintaining the typical infrastructure needed for application development and deployment.'

Reviewer: ```suggestion
Description: 'Function as a service is a cloud computing model that allows applications to be made without the infrastructure needed for development and deployment.'
```

The `Description` seemed a bit lengthy; this suggestion brings the character-count to exactly 150 characters.
